src/api/server.tsx

```diff
@@ -36,7 +36,29 @@ export const fetchResources = (): Promise<LxdResources> => {
   });
 };
 
-export const fetchConfigOptions = (): Promise<LxdConfigOptions> => {
+export const fetchConfigOptions = (
+  hasMetadataConfiguration: boolean,
+): Promise<LxdConfigOptions> => {
+  if (!hasMetadataConfiguration) {
+    return new Promise((resolve) =>
+      resolve({
+        configs: {
+          cluster: {},
+          instance: {},
+          project: {},
+          server: {},
+          "storage-btrfs": {},
+          "storage-ceph": {},
+          "storage-cephfs": {},
+          "storage-cephobject": {},
+          "storage-dir": {},
+          "storage-lvm": {},
+          "storage-zfs": {},
+        },
+      }),
```

Comment:
```suggestion
      resolve(null),
```
I think we can return null here and change the return type. All callers should be fine with this.